python/taichi/aot/module.py

```diff
@@ -1,4 +1,44 @@
-from taichi.lang import impl, kernel_arguments, kernel_impl
+from contextlib import contextmanager
+
+from taichi.lang import expr, impl, kernel_arguments, kernel_impl, matrix
+
+
+class KernelTemplate:
+    def __init__(self, kernel_fn, aot_module):
+        self._kernel_fn = kernel_fn
+        self._aot_module = aot_module
+
+    def instantiate(self, **kwargs):
+        name = self._kernel_fn.__name__
+        kernel = self._kernel_fn._primal
+        assert isinstance(kernel, kernel_impl.Kernel)
+        injected_args = []
+        key_p = ""
+        anno_index = 0
+        template_args = {}
+
+        for index, (key, value) in enumerate(kwargs.items()):
+            template_args[index] = (key, value)
+
+        for i in range(len(kernel.argument_annotations)):
+            anno = kernel.argument_annotations[i]
+            if isinstance(anno, kernel_arguments.template):
+                (k, v) = template_args[anno_index]
+                key_p += k
```

Comment:
Don't we deliminate primitive types? What happens if we `instantiate(a=1, b=False)`?